### PR TITLE
Add cli flag to only export completed tasks

### PR DESCRIPTION
**Pull request by adius**

### CODE REVIEW

**Comment by Thor77** [comment]
I don't really see the usecase for that in the export-subcommand, can you elaborate why you think this feature is useful?

File: wunderline-export.js [hunk 1]
```diff
@@ -35,29 +36,32 @@ function getLists (callback) {
...
       function getSubtasks (cb) {
-        api({url: '/subtasks', qs: {list_id: list.id}}, function (err, res, body) {
+        api({url: '/subtasks', qs: queryOptions}, function (err, res, body) {
           showProgress()
           cb(err, body)
         })
       },
       function getNotes (cb) {
-        api({url: '/notes', qs: {list_id: list.id}}, function (err, res, body) {
+        api({url: '/notes', qs: queryOptions}, function (err, res, body) {
```
Comment (Thor77) [inline wunderline-export.js:105]: `completed` is not a valid option for `/notes`

File: wunderline-export.js [hunk 2]
```diff
@@ -35,29 +36,32 @@ function getLists (callback) {
...
           if (err) process.exit(1)
-
           showProgress()
           cb(err, body)
         })
       },
       function getFiles (cb) {
-        api({url: '/files', qs: {list_id: list.id}}, function (err, res, body) {
+        api({url: '/files', qs: queryOptions}, function (err, res, body) {
```
Comment (Thor77) [inline wunderline-export.js:115]: `completed` is not a valid option for `/files`

**Review by Thor77 — CHANGES_REQUESTED** [review]
Don't add the `completed`-option to endpoints not supporting it.

File: wunderline-export.js (hunk 1 above)
Comment (adius) [reply wunderline-export.js:105]: Oh … good catch!

**Comment by adius** [comment]
> I don't really see the usecase for that in the export-subcommand

😳 I was really surprised that it only exported the open tasks.
If I export data from a program I expect it to export *all* data it has.
So I actually I would per default export the open *and* the completed tasks
and add `--open` and `--completed` flags if one wants to only export a subset.

**Comment by adius** [comment]
If you export your emails from your email app you don't expect it to only export your unread emails either, would you?  ^^

**Comment by Thor77** [comment]
> So I actually I would per default export the open and the completed tasks
and add --open and --completed flags if one wants to only export a subset.

Yeah, that sounds like the right way to do it, feel free to open another PR for the additional flag and changing the default output (or include it into this one).
Therefore I will merge your PR once you fix the two things mentioned in my review.

**Comment by Thor77** [comment]
Is there anything I can help you with to get this PR ready for merge @adius ?

**Comment by adius** [comment]
Not sure if this is the most beautiful solution,
but the limited Wunderlist API doesn't make things easy…

**Comment by Thor77** [comment]
Thanks for taking a look again and implementing the additional flag right away.
I'm not sure about changing the default, though. What do you think, @wayneashleyberry ?

**Comment by wayneashleyberry** [comment]
From my side I'd leave these filters out of the `export` command. Export was only ever intended to give you _all_ of your data, what you then choose to do with it is up to you.

There are many tools available such as [jid](https://github.com/simeji/jid) and [jq](https://stedolan.github.io/jq/) which provide excellent filtering for json.

Here is an example of filtering only your starred tasks from a full export using jq.

```sh
wunderline export | jq '.data.lists[].tasks[] | select(.starred == true)'
```

Theoretically a similar filter could be applied to get completed or uncompleted tasks, but I deleted all of my completed tasks haha

```sh
wunderline export | jq '.data.lists[].tasks[] | select(.completed == true)'
```

**Comment by wayneashleyberry** [comment]
Hey @adius, I'm going to close this pull request. I appreciate you pointing out that completed tasks weren't being exported, but I've opted for a simpler implementation.

**Comment by wayneashleyberry** [comment]
Merged in https://github.com/wayneashleyberry/wunderline/pull/105 and published as [v4.4.4](https://github.com/wayneashleyberry/wunderline/releases/tag/v4.4.4).

Please let me know if you're happy @adius. I've also updated the [readme](https://github.com/wayneashleyberry/wunderline#export) with examples for filtering out the exported data.

**Comment by adius** [comment]
Looks good, Thanks!

**Pull request closed without merging**